lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -130,6 +171,86 @@ registryUrlBuilder baseUrl pid _ (StakePoolMetadataHash bytes) =
     hashStr = T.unpack $ T.decodeUtf8 $ convertToBase Base16 bytes
     pidStr  = T.unpack $ toText pid
 
+-- | A smash GET request that reads the result at once into memory.
+smashRequest
+    :: Tracer IO StakePoolMetadataFetchLog
+    -> URI
+    -> Manager
+    -> ExceptT String IO ByteString
+smashRequest tr uri manager = getPayload
+  where
+    getPayload :: ExceptT String IO ByteString
+    getPayload = do
+        req <- withExceptT show $ except $ requestFromURI uri
+        liftIO $ traceWith tr $ MsgFetchSMASH uri
+        ExceptT
+            $ handle fromIOException
+            $ handle fromHttpException
+            $ withResponse req manager handleResponseStatus
+
+    handleResponseStatus response = case responseStatus response of
+        s | s == status200 -> do
+            let body = responseBody response
+            Right . BS.concat <$> brConsume body
+        s ->
+            pure $ Left $ mconcat
+                [ "The server replied with something unexpected: "
+                , show s
+                ]
+
+    fromHttpException :: Monad m => HttpException -> m (Either String a)
+    fromHttpException = return . Left . ("HTTP exception: " <>) . show
+
+data HealthCheck = HealthCheck
+    { status :: T.Text
+    , version :: T.Text
+    } deriving (Generic, Show, Eq, Ord)
+
+instance FromJSON HealthCheck where
+    parseJSON = genericParseJSON defaultRecordTypeOptions
+instance ToJSON HealthCheck where
+    toJSON = genericToJSON defaultRecordTypeOptions
+
+healthCheck
```

Comment:
Alright, so this is almost https://jira.iohk.io/browse/ADP-496 except a little ugly because https://jira.iohk.io/browse/ADP-502 is not done yet. I think it's harmless to keep this here. SMASH upstream has already merged the patch. I don't know how long it takes for our SMASH servers to be updated. Once they are, we can make this fatal and add it earlier in the call stack. Otherwise I can remove this commit.